openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -209,16 +224,59 @@ class AccessLevel:
     NO_ACCESS = None
 
 
-def list_libraries_for_user(user):
+def get_libraries_for_user(user):
     """
-    Lists up to 50 content libraries that the user has permission to view.
-
-    This method makes at least one HTTP call per library so should only be used
-    for development until we have something more efficient.
+    Return content libraries that the user has permission to view.
     """
     qs = ContentLibrary.objects.all()
-    filtered_qs = permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
-    return [get_library(ref.library_key) for ref in filtered_qs[:50]]
+    return permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
+
+
+def get_metadata_from_index(queryset):
+    """
+    Take a list of ContentLibrary objects and return metadata stored in
+    ContentLibraryIndex.
+    """
+    metadata = None
+    if ContentLibraryIndexer.indexing_is_enabled():
+        try:
+            library_keys = [lib.library_key for lib in queryset]
+            metadata = ContentLibraryIndexer.get_libraries(library_keys)
+        except (LibraryNotIndexedException, KeyError, ConnectionError) as e:
```

Comment:
The `KeyError` is a very broad catch, and might silence some unintended KeyErrors that we would prefer to be surfaced as 500s.

I assume that it's here to catch any library that is cached in an unexpected format, right? If so, consider catching the KeyError and then raising a custom exception.

Reply:
@kdmccormick True, I noticed this too and fixed this in #24510's PR [here](https://github.com/edx/edx-platform/pull/24510/commits/67f67ff7a18f49a2ae7e118516f6a1104bc5119c#diff-8cfdde6bceaaa7f5b1e1fbc912439f52L252). Let me know if I should backport those changes in this commit though for easier reviewing.

Reply:
@mavidser Sounds good. No need to backport anything.